src/features/page_view_timing/instrument/long-tasks.js

```diff
@@ -0,0 +1,60 @@
+import { subscribeToEOL } from '../../../common/unload/eol'
+
+/**
+ * Calls the `onReport` function for every entry reported by the PerformanceLongTaskTiming API.
+ * The reported value is a `DOMHighResTimeStamp`.
+ *
+ * The callback is always called when the page's visibility state changes to hidden.
+ * As a result, the `onReport` function might be called multiple times during the same page load.
+ *
+ * @param {Function} onReport - callback that accepts a `metric` object as the single parameter
+ */
+export const onLongTask = (onReport) => {
+  const handleEntries = (entries) => {
+    entries.forEach(entry => {
+      const metric = {
+        name: 'LT',
+        value: entry.duration,
+        info: {	// this property deviates from CWV std interface but will hold the custom context to send to NRDB
+          ltFrame: entry.name,				// MDN: the browsing context or frame that can be attributed to the long task
+          ltStart: entry.startTime,	// MDN: a double representing the time (millisec) when the task started
+          ltCtr: entry.attribution[0].containerType			// MDN: type of frame container: 'iframe', 'embed', or 'object' ... but this can also be 'window'
+        }
+      }
+      if (metric.info.ltCtr !== 'window') {	// the following properties are only of relevance & appended for html elements
+        Object.assign(metric.info, {
+          ltCtrSrc: entry.attribution[0].containerSrc,		// MDN: container's 'src' attribute
+          ltCtrId: entry.attribution[0].containerId,			// MDN: container's 'id' attribute
+          ltCtrName: entry.attribution[0].containerName		// MDN: container's 'name' attribute
+        })
+      }
+
+      onReport(metric)	// report every long task observed unconditionally
+    })
+  }
+
+  let observer
+  try {
+    if (PerformanceObserver.supportedEntryTypes.includes('longtask')) {
```

Comment:
Should we check if `PerformanceObserver` exists here? Thinking for IE11 support.

Reply:
Try block takes care of that; mirroring what core-vitals do so they don't bother with IE either (it's also excluded in PF tests).

Reply:
This feels like a code smell to me. We are using a `try..catch` instead of feature detection and the catch block is empty.

Reply:
I'd have to disagree there; it's pythonic in attempting to do something rather than waste time asking for permission. Both have merit, and I lean towards either depending on the situation. Ex., feature detect in metrics makes sense, and at the beginning of a whole module that assumes it, would be cleaner.

I think doing a check here would be tailored for IE, whereas this is expected to run just fine mostly, plus try-catching also includes any other exception, not just API support as a check does. Add to that it's consistent with PVT as a whole.